File: sort/Insertionsort.py
def Insertionsort(list):
    for i in range(len(list)):
        temp = list[i]
        j = i - 1
        while j >= 0 and temp < list[j]:
            list[j+1] = list[j]
            j = j - 1
        list[j+1] = temp
    return list

a = [3,4,2,0,8,9,1,7,6,5]
b = [5,2,4,6,5,7,3,1,2,0]
print (Insertionsort(a))
print (Insertionsort(b))

File: JAVA/basicAlgrithms/InsertionSort.java
package algorithm;

public class InsertionSort {
	public static int[] insertionsort(int[] array) {
		int index = 1;
		while (index < array.length) {
			for (int i = index; i > 0; i--) {
				if (array[i] < array[i - 1]) {
					int temp = array[i];
					array[i] = array[i - 1];
					array[i - 1] = temp;
				}
			}
			index++;
		}
		return array;
	}

	public static void main(String[] args) {
		int[] array = { 4, 5, 3, 7, 9, 5, 8, 11, -1 };
		array = insertionsort(array);
		for (int i : array) {
			System.out.print(i + ", ");
		}
	}
}
